tensorboard/components/tf_dashboard_common/tf-filterable-checkbox-list.html

```diff
@@ -58,7 +58,11 @@
           No matches
         </div>
       </template>
-      <template is="dom-repeat" items="[[_itemsMatchingRegex]]">
+      <template
+        is="dom-repeat"
+        items="[[_itemsMatchingRegex]]"
+        on-dom-change="_synchronizeColors"
```

Comment:
Is this deep / Does this trigger when users check/uncheck a checkbox?
If so, maybe consider an observer: `_synchronizeColors(_itemsMatchingRegex)`

Reply:
I do not follow. Why is that required?

Reply:
It seems it may be unnecessary to call `_synchronizeColors` on every single dom change.  It depends on the items, but not on whether a checkbox is checked.  If a user checks a checkbox and triggers dom to change, then we don't actually need to call it.

Reply:
> unnecessary to call _synchronizeColors on every single dom change.

this is the part I am not getting. `dom-change` is a feature of dom-repeat and dom-if. Are you saying paper-checkbox uses that and thus will fire bunch of events?

Reply:
> dom-change is a special event of dom-repeat and dom-if

Ahh, things make sense then, I was wrong!
For my own understanding: so before, when a component (not dom-repeat/if) has a `listeners: {'dom-change': ...}`, that means it is listening specifically for events dispatched by dom-repeat/if.